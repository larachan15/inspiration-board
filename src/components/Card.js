import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  render() {
    return (
      <div className="card">
        <div className="card__content">
          <p className="card__content-text">
            {this.props.text}
          </p>
          <p className="card__content-emoji">
            {this.props.emoji}
          </p>
        </div>
      </div>
    )
  }
}
// const Card = (props) => {
//   const { text, emoji } = props;
//
//   return (
//         <div className="card">
//           <div className="card__content">
//           <p className="card__content-text">{text}</p>
//           <p className="card__content-emoji">{emoji}</p>
//           </div>
//         </div>
//       )
// }

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string
};

export default Card;
