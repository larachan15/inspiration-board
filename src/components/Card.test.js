import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing card snapshot', () => {
    const wrapper = shallow( <Card
      text="My message!"
      emoji="black_heart"
      deleteCardCallback={() => {} } />);
      expect(wrapper).toMatchSnapshot();
    });
  });
