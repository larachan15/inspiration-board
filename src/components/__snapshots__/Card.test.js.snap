// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing card snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <p
      className="card__content-text"
    >
      My message!
    </p>
    <p
      className="card__content-emoji"
    >
      black_heart
    </p>
    <button
      className="card__delete"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
